lang/py/src/avro/io.py

```diff
@@ -219,6 +234,52 @@ def read_double(self):
       ((ord(self.read(1)) & 0xffL) << 56))
     return STRUCT_DOUBLE.unpack(STRUCT_LONG.pack(bits))[0]
 
+  def read_decimal_from_bytes(self, precision, scale):
+    """
+    Decimal bytes are decoded as signed short, int or long depending on the
+    size of bytes.
+    """
+    datum = self.read(self.read_long())
+    if len(datum) == 2:
+      unpacked_datum = STRUCT_SIGNED_SHORT.unpack(datum)[0]
+    elif len(datum) == 4:
+      unpacked_datum = STRUCT_SIGNED_INT.unpack(datum)[0]
+    else:
+      unpacked_datum = STRUCT_SIGNED_LONG.unpack(datum)[0]
+
+    original_prec = getcontext().prec
+    getcontext().prec = precision
+    scaled_datum = Decimal(unpacked_datum) / Decimal(pow(10, scale))
+    getcontext().prec = original_prec
+    return scaled_datum
+
+  def read_decimal_from_fixed(self, precision, scale, size):
+    """
+    Decimal is encoded as fixed. Fixed instances are encoded using the
+    number of bytes declared in the schema.
+    """
+    datum = self.read(size)
+    unscaled_datum = 0
+    msb = struct.unpack('!b', datum[0])[0]
```

Comment:
(not a blocker): it'd be helpful either to add a comment mentioned the `unpack` function returns a tuple, or to use tuple unpacking instead of referencing indices.